db/helpers_test.go

```diff
@@ -0,0 +1,34 @@
+package db
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var re = regexp.MustCompile(`(?i)^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`)
+
+func TestGenerateID(t *testing.T) {
+	t.Run("should return a UUIDv4", func(t *testing.T) {
+		for i := 1; i <= 1000; i++ {
+			id := generateID()
+
+			t.Run(id+" is a valid UUIDv4", func(t *testing.T) {
+				assert.Regexp(t, re, id)
+			})
+		}
+	})
+
+	t.Run("Shouldn't clash", func(t *testing.T) {
+		setOfIDs := make(map[string]bool)
```

Comment:
If you know you are going to be creating `1000000` ids, it might be more efficient to make the map that size to start with
`setOfIDs := make(map[string]bool, 1000000)`
Not too fussed though as it's test code

Reply:
Cheers, good to know 😄